test/compress/array-constructor.js

```diff
@@ -24,13 +24,25 @@ array_constructor_unsafe: {
         console.log(new Array(0));
         console.log(new Array(1));
         console.log(new Array(11));
+        console.log(Array(11));
         console.log(new Array(12));
+        console.log(Array(12));
+        console.log(new Array(foo));
+        console.log(Array(foo));
+        console.log(new Array("foo"));
+        console.log(Array("foo"));
     }
     expect: {
         console.log([]);
         console.log([]);
         console.log([,]);
         console.log([,,,,,,,,,,,]);
+        console.log([,,,,,,,,,,,]);
+        console.log(Array(12));
         console.log(Array(12));
+        console.log(Array(foo));
+        console.log(Array(foo));
+        console.log(Array("foo"));
+        console.log(Array("foo"));
```

Comment:
I think it would be good to optimize this too:
`Array(foo)` is equals to `[foo]`
`Array("Foo")` is equals to `["Foo"]`

Reply:
Let's not do it. Nobody creates arrays like that!